packages/preview/acrostiche-0.1.0/README.md

```diff
@@ -0,0 +1,51 @@
+# Acrostiche (0.1.0)
+
+Manages acronyms so you don't have to.
+
+## Usage
+
+The main goal of Acrostiche is to keep track of which acronym to define.
+
+### Define acronyms and proxy functions
+First, define the acronyms in a dictionary named `acronyms` with the keys being the acronyms, and the values their definition. If there is only a singular version of the definition, you can define it either as a string or as an array with a single string item. If there are both singular and plural versions, define the definition as an array where the first item is the singular definition and the second item is the plural.
+The first value of the array is the singular version of the definition. If there is a second value, it is the plural version of the definition. Any additional value is useless.
+
+Here is a sample of the `acronyms.typ` file:
+```
+#import "@preview/acrostiche:0.1.0": *
+
+#let acronyms = (
+  "OS": "Operating System", // Only singular, just a string
+  "NN": ("Neural Network",), // Only singular, array with single string item
+  "BIOS": ("Basic Input/Output System", "Basic Input/Output Systems"), // Singular and plural, array with singular first and plural second.
+)
+```
+Typst package system does not allow sharing variables between packages and the main document.
+Thus, you would have to pass the acronyms dictionary for each function call.
+However, because it would be annoying, you can simply copy-paste the following proxy functions at the top of your document to make it way easier. Just copy-paste and forget it!
+
+```
+#let get-def(acr,plural: false) = full-get-def(acr,plural,acronyms)
```

Comment:
I agree that Idealy there would not be anything to copy paste. However defining a variable in the main document and having it accessible to the package functions turned out more difficult than I thought. 
I don't want to make the functions call more complex (for example by passing the acronyms dictionary every time) because it slows down the writing.
I tried defining the acronyms dictionary in a state, but retrieving the dictionary from the state returns a None unless the dict is passed again as default value, which is the initial problem (if I had access to the initial value I would not bother making a state).
As far as I understand a factory functions would either be as cumbersome as the copy paste or make the writing very tedious as the user would call a unique function and pass the acronym and the name of the intended function as parameter so the factory could make the proper call. So not good either.

All in all it is not perfect and will definitely trick some users that don't read manual (that's why I added useful error messages) but this is the best I could come up with. I am obviously open to ANY suggestion to improve practicality.

Reply:
I tried the state solution again, and I found a good solution that dramatically improves practicality. I thought I understood states but apparently not yet ^^. I'll commit it today or tomorrow.